lib/oli/publishing.ex

```diff
@@ -512,6 +512,8 @@ defmodule Oli.Publishing do
       {:error, %Ecto.Changeset{}}
   """
   def delete_publication(%Publication{} = publication) do
```

Comment:
While this function existed prior to this PR, I found no reference to actually using it within the codebase.
It would not work if there were any PublishedResources associated with it, so I extended this function to delete any of those resources as part of deleting a publication

Reply:
We could also do this in the db by adding a migration that updates the `publication` association in the `published_resources` table to have the `on_delete: :delete_all` option.

Reply:
What if we try to set the [on_delete option](https://hexdocs.pm/ecto/Ecto.Schema.html#has_many/3-options) in the publication table for the published resources ?

Reply:
I started down this path but got a `** (Postgrex.Error) ERROR 0A000 (feature_not_supported) cannot alter type of a column used by a view or rule` while running the migration.

Sure enough, there's a materialzed view called "part__mapping" that references these tables.

In the migration, I think I need to drop that view, update the foreign key definition, and then recreate the view to make this work, but I've never dealt with views in postgres and am not entirely sure if that's safe to do. Thoughts on that?

Reply:
Looking at priv/repo/migrations/20220317164007_fix_previous_migration.exs, there's also a database trigger on the view which probably needs to be dropped/created as part of this. I think this migration would end up looking something like

```
Oli.Repo.Migrations.FixPreviousMigration.drop_trigger()
Oli.Repo.Migrations.FixPreviousMigration.drop_materialized_view()

drop the old constraint
add the new constraint with the on_delete option

Oli.Repo.Migrations.FixPreviousMigration.user = get_current_db_user()
Oli.Repo.Migrations.FixPreviousMigration.create_materialized_view(user)
Oli.Repo.Migrations.FixPreviousMigration.create_trigger(user)
Oli.Repo.Migrations.FixPreviousMigration.refresh_materialized_view()
```

Does this sound like the right path?

Reply:
I see... I've never dealt with views in postgres either but seems like dropping and recreating the trigger/view is the way to go. I think you could also avoid the drop constraint line if you use the `from` option in [`modify/3`](https://hexdocs.pm/ecto_sql/Ecto.Migration.html#modify/3) (I believe you could do `from: references(:publications)`)

Anyway, I assumed that modifying this field was way easier 😅. I'm ok with leaving it as is if it gets too complicated.

Reply:
I put that in and it seems to work well, migration only took a few seconds locally.

https://github.com/Simon-Initiative/oli-torus/pull/2528/files#diff-76f59a86fa8b417205c670a829a79d79fd590797fe67a6b7203df23d96bd04a5